stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/navigation/ChatNavigatorImpl.kt

```diff
@@ -2,7 +2,7 @@ package com.getstream.sdk.chat.navigation
 
 import com.getstream.sdk.chat.navigation.destinations.ChatDestination
 
-internal class ChatNavigatorImpl(private val handler: ChatNavigationHandler) : ChatNavigator {
+public class ChatNavigatorImpl(private val handler: ChatNavigationHandler) : ChatNavigator {
```

Comment:
We're opening these impl classes up only for the deprecated copy of ChatUI in the `ui-components` module, right? Should they be `@InternalStreamChatApi` instead of simply `public`?

Reply:
I think we can move logic from `ChatNavigatorImpl` class to `ChatNavigator` interface, and convert it to an open class

Reply:
The `old-ui` ChatNavigatorImpl is internal now. The `ui-components` one doesn't have interface now so if the user wants to set their custom navigator to `ChatUI` they need to inherit from `ChatNavigator` class.

Reply:
Can we remove this class then?

Reply:
Is this class needed anymore?